README.md

```diff
@@ -13,175 +13,413 @@ Compartmentalization of host behavior hooks for JS.
-  localTZAHook(): string;
-
-  // determines if the fn is acting as an "eval" function
-  isDirectEvalHook(evalFunctionRef: any): boolean;
-
-  // prep for trusted types non-string
-  canCompileHook(source: any, {
-    evaluator: functionRef, // can be a value from isDirectEvalHook
-    isDirect?: boolean
-  }): boolean; // need to allow mimicing CSP including nonces
+  // inherit, when true, indicates that this compartment shares the globals, global
+  // contour, and evaluators of the surrounding compartment.
+  // This would interact with the SES `lockdown` function: after lockdown,
+  // a true would be incoherent and throw an exception.
+  // When false, the compartment's globalThis would be populated only
+  // with a subset of the language's intrinsics and the compartment's
+  // specialized evaluators.
+  // For this proposal to be fully specified, we would need to realize
+  // the concept of shared intrinsics in anticipation of Lockdown.
+  // In order for the module-loader Compartment to be useful for a
+  // Lockdown-Compartment shim, we must disallow vendors from including
+  // any shared intrinsics beyond those specified by the module loader proposal.
+  // Note: The champions do not like the name `inherit` so this will likely
+  // change in a future version.
+  inherit: boolean,
+
+  // Globals to copy onto this compartment's unique globalThis.
+  // Constructor options with globals and inherit: true would be incoherent and
+  // effect an exception.
+  globals: Object,
+
+  // The compartment uses the resolveHook to synchronously elevate
+  // an import specifier (as it appears in the source of a StaticModuleRecord
+  // or bindings array of a third-party StaticModuleRecord), to
+  // the corresponding full specifier, given the full specifier of the
+  // referring (containing) module.
+  // The full specifier is the memo key for the module.
+  // TODO This proposal does not yet account for import assertions,
+  // which evidence shows are actually import type specifiers as they were
+  // originally designed and may need to also participate in the memo key.
+  resolveHook: (importSpecifier: string, referrerSpecifier: string) => string,
+
+  // The compartment's load and import methods may require to load or initialize
+  // additional modules.
+  // If the compartment does not have the corresponding module on hand,
+  // it will first consult the moduleMap for a descriptor of the needed module.
+  moduleMap?: Record<string, ModuleDescriptor>,
+
+  // If the moduleMap does not contain a descriptor for a necessary module,
+  // it will then consult the moduleMapHook.
+  // The moduleMapHook is necessary for matching computed specifiers,
+  // like specifiers that match a prefix and have a corresponding
+  // module descriptor.
+  // The moduleMap and moduleMapHook are both synchronous.
+  moduleMapHook?: (fullSpecifier: string) => ModuleDescriptor?,
```

Comment:
It turns out that the Agoric-inspired `moduleMapHook` has the same signature and behavior as the Moddable-inspired `loadNowHook`. It might be an overall improvement in names if we:

- renamed `moduleMap` to simply `modules`,
- renamed `moduleMapHook` to `loadNowHook`, and
- kept `loadHook`
- kept `importNow`

`moduleMap` was not awkward when it was a positional argument, but now the so-called module-map is a module-descriptor-dictionary (neither module, nor Map), and now is the time to consider bike-shedding.

Reply:
We could also alternately use the precedent from `Array.fromAsync` and:

- rename `moduleMap` to `modules`,
- rename `moduleMapHook` and `loadNowHook` to simply `loadHook`
- rename `loadHook` to `loadAsyncHook`

Reply:
I’m going to remove `moduleMapHook` for now, with the intention of leaving open the possibility of a TC53 carve-out for `loadNowHook` later. For general JavaScript (with promises), and with module descriptors supported on both sync on moduleMap (`modules`) and returned by `loadHook`, I think we have everything we need.